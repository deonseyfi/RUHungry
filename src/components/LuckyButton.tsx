import React from 'react';
import Button from '@material-ui/core/Button';
import { useNavigate } from 'react-router-dom';
import Emoji from './Emoji';
import CATEGORIES from '../App/constants';

/**
 * I'm Feeling Lucky Button.
 *
 * @returns {React.ReactElement} Button component.
 */
const LuckyButton = (): React.ReactElement => {
    const navigate = useNavigate();
    /**
     * Function to return a random food category from list.
     *
     * @returns {string} The randomly chosen food category.
     */
    const selectRandomCategory = (): string => {
        const categoryValue = CATEGORIES.map((category) => category.label);
        const randomCategory = Math.floor(Math.random() * categoryValue.length);
        // If returned category was All or blank, generate a new category.
        if (categoryValue[randomCategory] === 'All' || categoryValue[randomCategory] === '') {
            return selectRandomCategory();
        }
        return categoryValue[randomCategory];
    };
    return (
        <Button
            variant='outlined'
            color='inherit'
            startIcon={<Emoji label='shrug' symbol='ðŸ¤·' />}
            onClick={() => {
                navigate(`/restaurants/first=${selectRandomCategory()}`);
            }}
        >
            I&apos;m Feeling Lucky
        </Button>
    );
};

export default LuckyButton;
