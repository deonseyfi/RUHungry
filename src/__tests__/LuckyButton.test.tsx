import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LuckyButton from '../components/LuckyButton';

configure({ adapter: new Adapter() });

describe('LuckyButton test', () => {
    test('Renders LuckyButton component', () => {
        const wrapper = mount(<LuckyButton />);
        expect(wrapper).toMatchSnapshot();
    });
    test.skip('Selects random category', () => {
        const wrapper = mount(<LuckyButton />);
        const random = jest.spyOn(LuckyButton.prototype, 'selectRandomCategory');
        wrapper.simulate('click');
        expect(random).toHaveBeenCalled();
    });
    test.skip('Click LuckyButton', () => {
        const wrapper = mount(<LuckyButton />);
        const random = jest.spyOn(LuckyButton.prototype, 'selectRandomCategory');
        wrapper.simulate('click');
        expect(random).toHaveBeenCalled();
    });
});
