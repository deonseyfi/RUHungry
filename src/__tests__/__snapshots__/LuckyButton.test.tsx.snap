// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LuckyButton test Renders LuckyButton component 1`] = `ReactWrapper {}`;
